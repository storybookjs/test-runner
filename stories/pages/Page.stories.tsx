import type { Meta, StoryObj } from '@storybook/react-vite';
import { within, userEvent } from 'storybook/test';

import { Page } from './Page';

const meta = {
  title: 'Pages/page',
  component: Page,
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof Page>;

export default meta;
type Story = StoryObj<typeof meta>;

export const LoggedOut: Story = {};

export const LoggedIn: Story = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const loginButton = await canvas.getByRole('button', {
      name: /Log in/i,
    });
    await userEvent.click(loginButton);
  },
};
