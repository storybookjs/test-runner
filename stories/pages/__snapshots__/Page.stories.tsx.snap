// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pages/page LoggedIn play-test 2`] = `
<article>
  <header>
    <div class="storybook-header">
      <div>
        <svg width="32"
             height="32"
             viewbox="0 0 32 32"
             xmlns="http://www.w3.org/2000/svg"
        >
          <g fill="none"
             fill-rule="evenodd"
          >
            <path d="M10 0h12a10 10 0 0110 10v12a10 10 0 01-10 10H10A10 10 0 010 22V10A10 10 0 0110 0z"
                  fill="#FFF"
            >
            </path>
            <path d="M5.3 10.6l10.4 6v11.1l-10.4-6v-11zm11.4-6.2l9.7 5.5-9.7 5.6V4.4z"
                  fill="#555AB9"
            >
            </path>
            <path d="M27.2 10.6v11.2l-10.5 6V16.5l10.5-6zM15.7 4.4v11L6 10l9.7-5.5z"
                  fill="#91BAF8"
            >
            </path>
          </g>
        </svg>
        <h1>
          Acme
        </h1>
      </div>
      <div>
        <span class="welcome">
          Welcome,
          <b>
            Jane Doe
          </b>
          !
        </span>
        <button type="button"
                class="storybook-button storybook-button--small storybook-button--secondary"
        >
          Log out
        </button>
      </div>
    </div>
  </header>
  <section class="storybook-page">
    <h2>
      Pages in Storybook
    </h2>
    <p>
      We recommend building UIs with a
      <a href="https://componentdriven.org"
         target="_blank"
         rel="noopener noreferrer"
      >
        <strong>
          component-driven
        </strong>
      </a>
      process starting with atomic components and ending with pages.
    </p>
    <p>
      Render pages with mock data. This makes it easy to build and review page states without needing to navigate to them in your app. Here are some handy patterns for managing page data in Storybook:
    </p>
    <ul>
      <li>
        Use a higher-level connected component. Storybook helps you compose such data from the "args" of child component stories
      </li>
      <li>
        Assemble data in the page component from your services. You can mock these services out using Storybook.
      </li>
    </ul>
    <p>
      Get a guided tutorial on component-driven development at
      <a href="https://storybook.js.org/tutorials/"
         target="_blank"
         rel="noopener noreferrer"
      >
        Storybook tutorials
      </a>
      . Read more in the
      <a href="https://storybook.js.org/docs"
         target="_blank"
         rel="noopener noreferrer"
      >
        docs
      </a>
      .
    </p>
    <div class="tip-wrapper">
      <span class="tip">
        Tip
      </span>
      Adjust the width of the canvas with the
      <svg width="10"
           height="10"
           viewbox="0 0 12 12"
           xmlns="http://www.w3.org/2000/svg"
      >
        <g fill="none"
           fill-rule="evenodd"
        >
          <path d="M1.5 5.2h4.8c.3 0 .5.2.5.4v5.1c-.1.2-.3.3-.4.3H1.4a.5.5 0 01-.5-.4V5.7c0-.3.2-.5.5-.5zm0-2.1h6.9c.3 0 .5.2.5.4v7a.5.5 0 01-1 0V4H1.5a.5.5 0 010-1zm0-2.1h9c.3 0 .5.2.5.4v9.1a.5.5 0 01-1 0V2H1.5a.5.5 0 010-1zm4.3 5.2H2V10h3.8V6.2z"
                id="a"
                fill="#999"
          >
          </path>
        </g>
      </svg>
      Viewports addon in the toolbar
    </div>
  </section>
</article>
`;

exports[`Pages/page LoggedOut smoke-test 2`] = `
<article>
  <header>
    <div class="storybook-header">
      <div>
        <svg width="32"
             height="32"
             viewbox="0 0 32 32"
             xmlns="http://www.w3.org/2000/svg"
        >
          <g fill="none"
             fill-rule="evenodd"
          >
            <path d="M10 0h12a10 10 0 0110 10v12a10 10 0 01-10 10H10A10 10 0 010 22V10A10 10 0 0110 0z"
                  fill="#FFF"
            >
            </path>
            <path d="M5.3 10.6l10.4 6v11.1l-10.4-6v-11zm11.4-6.2l9.7 5.5-9.7 5.6V4.4z"
                  fill="#555AB9"
            >
            </path>
            <path d="M27.2 10.6v11.2l-10.5 6V16.5l10.5-6zM15.7 4.4v11L6 10l9.7-5.5z"
                  fill="#91BAF8"
            >
            </path>
          </g>
        </svg>
        <h1>
          Acme
        </h1>
      </div>
      <div>
        <button type="button"
                class="storybook-button storybook-button--small storybook-button--secondary"
        >
          Log in
        </button>
        <button type="button"
                class="storybook-button storybook-button--small storybook-button--primary"
        >
          Sign up
        </button>
      </div>
    </div>
  </header>
  <section class="storybook-page">
    <h2>
      Pages in Storybook
    </h2>
    <p>
      We recommend building UIs with a
      <a href="https://componentdriven.org"
         target="_blank"
         rel="noopener noreferrer"
      >
        <strong>
          component-driven
        </strong>
      </a>
      process starting with atomic components and ending with pages.
    </p>
    <p>
      Render pages with mock data. This makes it easy to build and review page states without needing to navigate to them in your app. Here are some handy patterns for managing page data in Storybook:
    </p>
    <ul>
      <li>
        Use a higher-level connected component. Storybook helps you compose such data from the "args" of child component stories
      </li>
      <li>
        Assemble data in the page component from your services. You can mock these services out using Storybook.
      </li>
    </ul>
    <p>
      Get a guided tutorial on component-driven development at
      <a href="https://storybook.js.org/tutorials/"
         target="_blank"
         rel="noopener noreferrer"
      >
        Storybook tutorials
      </a>
      . Read more in the
      <a href="https://storybook.js.org/docs"
         target="_blank"
         rel="noopener noreferrer"
      >
        docs
      </a>
      .
    </p>
    <div class="tip-wrapper">
      <span class="tip">
        Tip
      </span>
      Adjust the width of the canvas with the
      <svg width="10"
           height="10"
           viewbox="0 0 12 12"
           xmlns="http://www.w3.org/2000/svg"
      >
        <g fill="none"
           fill-rule="evenodd"
        >
          <path d="M1.5 5.2h4.8c.3 0 .5.2.5.4v5.1c-.1.2-.3.3-.4.3H1.4a.5.5 0 01-.5-.4V5.7c0-.3.2-.5.5-.5zm0-2.1h6.9c.3 0 .5.2.5.4v7a.5.5 0 01-1 0V4H1.5a.5.5 0 010-1zm0-2.1h9c.3 0 .5.2.5.4v9.1a.5.5 0 01-1 0V2H1.5a.5.5 0 010-1zm4.3 5.2H2V10h3.8V6.2z"
                id="a"
                fill="#999"
          >
          </path>
        </g>
      </svg>
      Viewports addon in the toolbar
    </div>
  </section>
</article>
`;
